tests/ConnectionTest.php

```diff
@@ -396,6 +396,48 @@ public function testConnectStartsTransactionInNoAutoCommitMode(): void
         self::assertTrue($conn->isTransactionActive());
     }
 
+    public function testTransactionIsNotActiveAfterTransactionalInAutoCommitMode(): void
+    {
+        $driverMock = $this->createStub(Driver::class);
+        $driverMock
+            ->method('connect')
+            ->willReturn(
+                $this->createMock(DriverConnection::class),
+            );
+
+        $conn = new Connection([], $driverMock);
+
+        $conn->setAutoCommit(true);
+
+        self::assertFalse($conn->isTransactionActive());
+
+        $conn->transactional(static function (Connection $connection): void {
+        });
+
+        self::assertFalse($conn->isTransactionActive());
+    }
+
+    public function testTransactionIsActiveAfterTransactionalInNoAutoCommitMode(): void
+    {
+        $driverMock = $this->createMock(Driver::class);
+        $driverMock
+            ->method('connect')
+            ->willReturn(
+                $this->createMock(DriverConnection::class),
+            );
+
+        $conn = new Connection([], $driverMock);
+
+        $conn->setAutoCommit(false);
+
+        self::assertFalse($conn->isTransactionActive());
```

Comment:
This assertion is very misleading.

The fact that there's no transaction active has nothing to do with the auto commit setting. We're simply not connected to the database because the connection is lazy.

Reply:
I'll simply remove the line?

Reply:
That would solve my problem, yes.